components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/events/TokenRevocationNotifierImpl.java

```diff
@@ -0,0 +1,156 @@
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.keymgt.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.keymgt.token.TokenRevocationNotifier;
+import org.wso2.carbon.databridge.commons.Event;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Implemented class for TokenRevocationNotifier interface
+ * SendMessageOnRealtime() method is implemented
+ * SendMessageToPersistentStorage() method is implemented
+ */
+public class TokenRevocationNotifierImpl implements TokenRevocationNotifier {
+
+    private static final Log log = LogFactory.getLog(APIMOAuthEventInterceptor.class);
+
+    //Variables related to Realtime Notifier
+    private static String realtimeNotifierTTL;
+
+    //Variables related to Persistent Notifier
+    private static String persistentNotifierHostname;
+    private static String persistentNotifierTTL;
+    private static String persistentNotifierUsername;
+    private static String persistentNotifierPassword;
+
+    private Map<String, Properties> tokenRevocationNotifiers;
+
+    public TokenRevocationNotifierImpl(Map<String, Properties> tokenRevocationNotifiers) {
+
+        this.tokenRevocationNotifiers = tokenRevocationNotifiers;
+        extractNotifiers();
+
+    }
+
+    /**
+     * Method to extract the config properties of revoke token notifiers
+     */
+    private void extractNotifiers() {
+
+        //Default Constant Values
+        final String DEFAULT_PERSISTENT_NOTIFIER_HOSTNAME = "https://localhost:2379/v2/keys/jti/";
+        final String DEFAULT_TTL = "3600";
+        final String DEFAULT_PERSISTENT_NOTIFIER_USERNAME = "root";
+        final String DEFAULT_PERSISTENT_NOTIFIER_PASSWORD = "root";
+
+        for (Map.Entry<String, Properties> entry : tokenRevocationNotifiers.entrySet()) {
+            if (entry.getKey().equalsIgnoreCase("RealtimeNotifier")) {
+                Properties properties = entry.getValue();
+                realtimeNotifierTTL = properties.getProperty("ttl", DEFAULT_TTL);
+            }
+            if (entry.getKey().equalsIgnoreCase("PersistentNotifier")) {
+                Properties properties = entry.getValue();
+                persistentNotifierHostname = properties.getProperty("hostname", DEFAULT_PERSISTENT_NOTIFIER_HOSTNAME);
+                persistentNotifierTTL = properties.getProperty("ttl", DEFAULT_TTL);
+                persistentNotifierUsername = properties.getProperty("username", DEFAULT_PERSISTENT_NOTIFIER_USERNAME);
+                persistentNotifierPassword = properties.getProperty("password", DEFAULT_PERSISTENT_NOTIFIER_PASSWORD);
+            }
+        }
+    }
+
+    /**
+     * Method to publish the revoked token on to the realtime message broker
+     *
+     * @param revokedToken requested revoked token
+     */
+    @Override public void
```

Comment:
shouldn't `public void` come to the next line?

Reply:
resolved